core/src/main/java/com/amplifyframework/core/model/query/Where.java

```diff
@@ -63,6 +69,38 @@ public static QueryOptions id(@NonNull final String modelId) {
                 .paginated(Page.firstResult());
     }
 
+    /**
+     * Factory method that builds the options with a predicate matching the primary key and the
+     * pagination set to the first result only.
+     *
+     * @param itemClass model class.
+     * @param modelPrimaryKey model identifier.
+     * @param <T> Extends Model.
+     * @return options with proper predicate and pagination to match a model by its id.
+     * @throws AmplifyException Throws AmplifyException.
+     */
+    public static <T extends Model> QueryOptions identifier(@NonNull Class<T> itemClass,
+                                                           @NonNull final Serializable modelPrimaryKey)
+            throws AmplifyException {
+        final ModelSchema schema = ModelSchema.fromModelClass(itemClass);
```

Comment:
This schema resolving doesn't support Flutter use case, as Flutter cannot provide `itemClass`.

Ideally, `identifier` should accept `modelName` along with `modelPrimaryKey`, then we can look up the model schema from  `SchemaRegistry` singleton.

Reply:
This is a convenience method created for Android. If flutter needs something like this we can create another implementation.